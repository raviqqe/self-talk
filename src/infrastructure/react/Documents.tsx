import { PulseLoader } from "react-spinners";
import InfiniteScroller from "react-infinite-scroller";
import { useAsync } from "react-use";
import React from "react";
import styled from "styled-components";
import { IDocument } from "../../domain/document";
import { InsertFilesFunction } from "./utilities";
import { Document } from "./Document";
import { white } from "./style/colors";

const Container = styled.div`
  display: flex;
  flex-direction: column-reverse;
  overflow: auto;
`;

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledInfiniteScroller = styled(InfiniteScroller)`
  display: flex;
  flex-direction: column-reverse;
  padding: 1em 0.5em;
`;

const StyledDocument = styled(Document)`
  margin: 0.5em;
`;

export interface IProps {
  documents: IDocument[] | null;
  insertFiles: InsertFilesFunction;
  listDocuments: () => Promise<void>;
  listMoreDocuments: () => Promise<void>;
  updateDocument: (document: IDocument) => Promise<void>;
}

export const Documents = ({
  documents,
  insertFiles,
  listDocuments,
  listMoreDocuments,
  updateDocument,
}: IProps) => {
  useAsync(listDocuments, []);

  return documents ? (
    <Container>
      <StyledInfiniteScroller
        hasMore={true}
        isReverse={true}
        loadMore={listMoreDocuments}
        threshold={512}
        useWindow={false}
      >
        {documents.map((document: IDocument) => (
          <StyledDocument
            document={document}
            insertFiles={insertFiles}
            key={document.id}
            updateDocument={updateDocument}
          />
        ))}
      </StyledInfiniteScroller>
    </Container>
  ) : (
    <LoaderContainer>
      <PulseLoader color={white} />
    </LoaderContainer>
  );
};
