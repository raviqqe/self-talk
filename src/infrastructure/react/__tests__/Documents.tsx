import { create } from "react-test-renderer";
import { Documents } from "../Documents";

it("renders", () => {
  expect(
    create(
      <Documents
        documents={[{ id: "id", text: "text" }]}
        insertFiles={async () => ""}
        listDocuments={async () => {}}
        listMoreDocuments={async () => {}}
        updateDocument={async () => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("renders with no documents", () => {
  expect(
    create(
      <Documents
        documents={[]}
        insertFiles={async () => ""}
        listDocuments={async () => {}}
        listMoreDocuments={async () => {}}
        updateDocument={async () => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("renders with documents not loaded yet", () => {
  expect(
    create(
      <Documents
        documents={null}
        insertFiles={async () => ""}
        listDocuments={async () => {}}
        listMoreDocuments={async () => {}}
        updateDocument={async () => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
