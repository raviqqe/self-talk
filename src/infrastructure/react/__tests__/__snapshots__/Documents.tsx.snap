// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-fzoLsD bYuRlV"
>
  <div
    className="sc-fzplWN jkWxXU"
  >
    <div
      className="sc-AxmLO ehOlxA sc-fznyAO yfPnD"
    >
      <div
        className="sc-AxirZ hvfzc"
      >
        <p>
          text
        </p>
      </div>
      <div
        className="sc-fzozJi eyZSoi"
      >
        <div
          aria-label="Edit"
          className="sc-AxjAm ijbaQh"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with documents not loaded yet 1`] = `
<div
  className="sc-fzpans ciBlfK"
>
  <div
    className="css-0"
  >
    <div
      className="css-b0t2tw"
    />
    <div
      className="css-fz02ge"
    />
    <div
      className="css-1xlchjq"
    />
  </div>
</div>
`;

exports[`renders with no documents 1`] = `
<div
  className="sc-fzoLsD bYuRlV"
>
  <div
    className="sc-fzplWN jkWxXU"
  />
</div>
`;
